idx/cassandra/cassandra.go

```diff
@@ -390,6 +393,9 @@ func (c *CasIdx) Delete(orgId int, pattern string) ([]schema.MetricDefinition, e
 }
 
 func (c *CasIdx) deleteDef(def *schema.MetricDefinition) error {
+	if !updateCassIdx {
+		return nil
+	}
```

Comment:
I think checks like these should be executed in the caller, e.g. in the public functions, not the internal "worker" functions. This is also more efficient. E.g. In Prune we can simply skip the entire loop of many deleteDef calls.